File: 2018_Graduation_project#convert_to_java/TestApp/app/src/main/java/com/android/test/PageAdapter.java
package com.android.test;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.PagerAdapter;

import java.util.ArrayList;

public final class PageAdapter extends FragmentStatePagerAdapter{

    private final DatabaseHelper DB;
    private ArrayList tabNameArray;
    private final Context context;
    public final DatabaseHelper getDB() {
        return this.DB;
    }
    public final ArrayList getTabNameArray() {
        return this.tabNameArray;
    }
    public final void setTabNameArray(ArrayList var1) { this.tabNameArray = var1; }
    public int getItemPosition(Object object) { return PagerAdapter.POSITION_NONE; }
    public Fragment getItem(int TabId) {
        return (Fragment)new MenuListFragment(context, (String)tabNameArray.get(TabId));
    }
    public int getCount() {
        this.tabNameArray = this.DB.FetchTabName();
        return this.tabNameArray.size();
    }
    public CharSequence getPageTitle(int position) {
        if(position<0)
            return null;
        return (CharSequence)this.tabNameArray.get(position);
    }
    public PageAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
        this.DB = new DatabaseHelper(this.context);
        this.tabNameArray = this.DB.FetchTabName();
    }
}

File: 20180609_android_kotlin/app/src/main/java/com/example/master/a20180609_android_kotlin/PageAdapter.kt
package com.example.master.a20180609_android_kotlin

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.view.PagerAdapter

class PageAdapter(fm: FragmentManager, private val context: Context) : FragmentStatePagerAdapter(fm) {

    val DB = DatabaseHelper(context)
    var tabNameArray = DB.FetchTabName()

    override fun getItemPosition(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun getItem(TabId : Int): Fragment {
        return MenuListFragment.newInstance(context,tabNameArray.get(TabId))
    }

    override fun getCount(): Int {
        tabNameArray = DB.FetchTabName()
        return tabNameArray.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return tabNameArray.get(position)
    }

}